allennlp/data/iterators/basic_iterator.py

```diff
@@ -32,14 +32,20 @@ class BasicIterator(DataIterator):
         If specified, the iterator will load this many instances at a time into an
         in-memory list and then produce batches from one such list at a time. This
         could be useful if your instances are read lazily from disk.
+    maximum_samples_per_batch : tuple(padding_key, limit), optional, (default = None)
```

Comment:
this type hint should be : `Tuple[str, int]` and then describe them in text after.

Reply:
This is fixed in the latest commits.